crates/core/src/host/scheduler.rs

```diff
@@ -327,72 +331,89 @@ impl SchedulerActor {
             return;
         };
         let db = module_host.replica_ctx().relational_db.clone();
-        let module_host_clone = module_host.clone();
 
-        let res = tokio::spawn(async move { module_host.call_scheduled_reducer(item).await }).await;
+        // Determine the call params.
+        // This also lets us know whether to call a reducer or procedure.
+        let params = call_params_for_queued_item(&module_host.info, &db, item);
```

Comment:
The current code fetches the parameters and executes the reducer within the same `tx` for atomicity.
I think this leaves the room for `params` to be modified by the item they are used for executeion.